src/Infrastructure/Extensions/NServiceBusExtensions.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Azure.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using System.Text.RegularExpressions;
+
+namespace SFA.DAS.Apprenticeships.Infrastructure.Extensions;
+
+public static class NServiceBusExtensions
+{
+    public static void ConfigureNServiceBusForSend(this IServiceCollection services, string fullyQualifiedNamespace)
+    {
+        var endpointConfiguration = new EndpointConfiguration("SFA.DAS.Funding.ApprenticeshipEarnings");
+        endpointConfiguration.UseSerialization<SystemJsonSerializer>();
+        endpointConfiguration.SendOnly();
+
+        var transport = endpointConfiguration.UseTransport<AzureServiceBusTransport>();
+        transport.CustomTokenCredential(fullyQualifiedNamespace, new DefaultAzureCredential());
+        endpointConfiguration.Conventions().SetConventions();
+
+        var endpointInstance = Endpoint.Start(endpointConfiguration).GetAwaiter().GetResult();
+        services.AddSingleton<IMessageSession>(endpointInstance);
+    }
+
+    public static void SetConventions(this ConventionsBuilder conventions)
+    {
+        conventions.DefiningEventsAs(IsEvent);
+        conventions.DefiningCommandsAs(t => false);
+        conventions.DefiningMessagesAs(t => false);
+    }
+
+    public static string GetFullyQualifiedNamespace(this string serviceBusConnectionString)
```

Comment:
I know this was done previously when we were using the shared package, but is this needed? What are the variables for service bus connection string set to in the envs? if it's just the fully qualified namespace and we can use the same locally maybe we don't need to do this stripping of the connection string?

Reply:
My question is are both us locally and devops on deployment are setting the connection string to simply the fully qualified namespace? If so we don't need to strip the connection string. This method isn't making the connection it's just doing string manipulation on the variable.